e2e/dractions/crud.go

```diff
@@ -230,6 +236,10 @@ func generateDRPCDiscoveredApps(name, namespace, clusterName, drPolicyName, plac
 		},
 	}
 
+	if recipeRef != nil {
+		drpc.Spec.KubeObjectProtection.RecipeRef = recipeRef
+	}
```

Comment:
Why not set it always? if the argument is nil setting it to nil is correct, right?